modal/functions.py

```diff
@@ -277,7 +282,160 @@ class _Function(_Object, type_prefix="fu"):
     _tag: str
     _raw_f: Callable[..., Any]
     _build_args: dict
-    _parent: "_Function"
+    _can_use_base_function: bool = False  # whether we need to call FunctionBindParams
+
+    # when this is the method of a class/object function, invocation of this function
+    # should be using another function id and supply the method name in the FunctionInput:
+    _use_function_id: str  # The function to invoke
+    _use_method_name: str = ""
+
+    # TODO (elias): remove _parent. In case of instance functions, and methods bound on those,
+    #  this references the parent class-function and is used to infer the client for lazy-loaded methods
+    _parent: Optional["_Function"] = None
+
+    def _bind_method(
+        self,
+        user_cls,
+        method_name: str,
+        partial_function: "modal.partial_function._PartialFunction",
+    ):
+        """mdmd:hidden
+
+        Creates a function placeholder function that binds a specific method name to
+        this function for use when invoking the function.
```

Comment:
Somewhat unfortunate that we're overloading "bind" terminology with classes since we already have the `FunctionBindParams` concept, which I'm pretty sure is mostly unrelated to what's happening here?

Reply:
Yeah there is a two-dimensional "bind" on class functions now:
* class parameters (using FunctionBindParams to create an "instance service function" from a "class service function")
* method to use (which is done here)